matchzoo/embedding.py

```diff
@@ -1,143 +1,68 @@
 
+import abc
+import csv
+import typing
 
-logger = logging.getLogger(__name__)
+import numpy as np
+import pandas as pd
 
 
 class Embedding(object):
     """
     Embedding class.
-
-    TODO: https://github.com/dmlc/gluon-nlp/blob/
-          5e65c4751e1be8920b1021822a988af77224902f/
-          gluonnlp/embedding/token_embedding.py
-
-    Examples:
-        >>> import matchzoo as mz
-        >>> embed = mz.datasets.toy.load_embedding()
-        >>> # Need term_index
-        >>> embed.build({'G': 1, 'C': 2, 'D': 3, 'A': 4, '[PAD]': 0})
-        >>> index_state = embed.index_state
-        >>> index_state # doctest: +SKIP
-        {4: 1, 2: 1, 3: 1, 1: 2, 0: 0}
-        >>> embedding_mat = embed.embedding_mat
-        >>> embedding_mat # doctest: +SKIP
-        array([[ 0.       ,  0.       ,  0.       ,  0.       ,  0.        ,
-             0.        ,  0.        ,  0.        ,  0.        ,  0.        ],
-           [ 0.22785383, -0.09405118,  0.15446669,  0.20727136,  0.05943427,
-             0.12673594,  0.02788511, -0.0433806 , -0.20548974,  0.24532762],
-           [ 0.1       ,  0.2       ,  0.3       ,  0.4       ,  0.5       ,
-             0.6       ,  0.7       ,  0.8       ,  0.9       ,  1.        ],
-           [ 0.1       ,  0.2       ,  0.3       ,  0.4       ,  0.5       ,
-             0.6       ,  0.7       ,  0.8       ,  0.9       ,  1.        ],
-           [ 0.1       ,  0.2       ,  0.3       ,  0.4       ,  0.5       ,
-             0.6       ,  0.7       ,  0.8       ,  0.9       ,  1.        ]])
-        >>> embedding_mat[0] # doctest: +SKIP
-        array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])
-
     """
 
-    def __init__(self, embedding_file):
-        """
-        Class initialization.
+    def __init__(self):
+        """Init."""
+        self._matrix = None
 
-        :param embedding_file: the embeding input file path.
-        """
-        self._embedding_file = embedding_file
-        self._embedding_dim = 0
-        self._index_state = {}
-        self._embedding_mat = None
+    def _entity_to_vec(self, entity: str):
+        """Get word embedding by entity."""
+        return self._matrix.loc[entity].values
 
-    @property
-    def embedding_file(self) -> str:
-        """Get embedding file name."""
-        return self._embedding_file
+    def __getitem__(self, entities: typing.Union[str, list]):
+        """Get embeddings by entity of list of entities."""
+        if isinstance(entities, str):
+            return self._entity_to_vec(entities)
 
-    @property
-    def embedding_dim(self) -> int:
-        """Get embedding dimension."""
-        return self._embedding_dim
+        return np.vstack([self._entity_to_vec(entity) for entity in entities])
 
     @property
-    def index_state(self) -> dict:
-        """
-        Get word index state dictionary.
-
-        0 - [PAD] word, 1 - found embed, 2 - OOV word.
-        """
-        return self._index_state
+    def dimension(self):
```

Comment:
I suggest renaming this to `output_dim` while also adding another property `input_dim` just to be consistent with namings in `BaseModel.get_default_params(with_embedding=True)`.